### PR TITLE
Improve lists

### PR DESCRIPTION
**Pull request by AntonPuko**
Closes https://github.com/Brewskey/Brewskey.App/issues/37
Closes https://github.com/Brewskey/Brewskey.App/issues/36

I know there is some duplication, but I still not sure if I want to create DAOEntityList at this point.

### CODE REVIEW

File: src/common/InfiniteLoader.js [hunk 1]
```diff
@@ -7,7 +7,7 @@ import { ActivityIndicator, Platform } from 'react-native';
 
 type Props = {|
   children?: Function,
-  fetchNextData: () => Promise<void>,
+  fetchNextData: () => Promise<*>,
```
Comment (jlkalberer) [inline src/common/InfiniteLoader.js]: This makes me think it should be `type Props<TType>`

File: src/common/SwipeableFlatList/index.js [hunk 2]
```diff
@@ -25,47 +16,26 @@ import { FlatList, View } from 'react-native';
...
-  renderQuickActions: renderItemType,
   preventSwipeRight?: boolean,
+  renderQuickActions: renderItemType,
 };
 
 type Props<TEntity> = SwipeableListProps & FlatListProps<TEntity>;
 
 type State = {
   openRowKey: ?string,
+  refreshing: boolean,
```
Comment (jlkalberer) [inline src/common/SwipeableFlatList/index.js]: Can this be `isRefreshing`?

File: src/stores/DAOEntityStore.js [hunk 3]
```diff
@@ -121,14 +121,36 @@ class DAOEntityStore<TEntity: { id: string }, TEntityMutator: {}> {
...
-  );
+  getByQueryOptions = createTransformer((queryOptions: QueryOptions): Array<
+    TEntity,
+  > => {
+    const { filters, orderBy } = queryOptions;
+    let entities = this.all;
+
+    if (filters) {
+      entities = entities.filter((entityItem: TEntity): boolean =>
+        DAOApi.doesSatisfyToQueryFilters(entityItem, filters),
```
Comment (jlkalberer) [inline src/stores/DAOEntityStore.js]: nit: `doesSatisfyToQueryFilters` seems weird.  Maybe `doesSatisfyQueryFilters`

File: src/stores/DAOEntityStore.js [hunk 4]
```diff
@@ -121,14 +121,36 @@ class DAOEntityStore<TEntity: { id: string }, TEntityMutator: {}> {
...
+      // it makes assumption that we use exactly id prop for ordering
+      // and id is exactly `kind of number` so we can parse it.
+      const { column, direction } = orderBy[0];
+
+      entities.sort(
+        (item: TEntity, nextItem: TEntity): number =>
+          parseInt(item[column], 10) - parseInt(nextItem[column], 10),
+      );
+
+      if (direction === 'desc') {
```
Comment (jlkalberer) [inline src/stores/DAOEntityStore.js]: Why not pass this as a parameter to OData?  Just reversing the entities will result an an incorrect dataset.

File: src/stores/DAOEntityStore.js (hunk 3 above)
Comment (AntonPuko) [reply src/stores/DAOEntityStore.js]: ok,  but its inside our api lib, so I'll have to switch there first.

File: src/stores/DAOEntityStore.js (hunk 4 above)
Comment (AntonPuko) [reply src/stores/DAOEntityStore.js]: its both, in odata and in store...In the store I'm basically trying to parse queryOptions which I use inside list to make it consistent...So queryOptions is one source of truth as filter for fetching and gettings items from the store,  If I don't do this, I can't use it as cachable entity store, I'll have to clear it every time for every list/whatever.

File: src/common/SwipeableFlatList/index.js (hunk 2 above)
Comment (AntonPuko) [reply src/common/SwipeableFlatList/index.js]: ofc, just FlatList native prop called `refreshing` so I did it for consistency, but like you want )

File: src/stores/DAOEntityStore.js (hunk 4 above)
Comment (jlkalberer) [reply src/stores/DAOEntityStore.js]: This doesn't make sense.  OData should always be the source of truth and however the data is stored should already match the query options.

Comment (jlkalberer) [reply src/stores/DAOEntityStore.js]: I also do not care about caching.  It seems like when you focus on this, it ends up making everything much more complicated.

**Pull request merged**